src/sessions.js

```diff
@@ -26,9 +26,21 @@
 					}
 					saved = false; // it may be safe, sure, but you haven't "Saved" it
 				});
+			} else {
+				// no uri so lets save the blank canvas
+				storage.set(lsid, canvas.toDataURL("image/png"), function(err){
+					if(err){
+						if(err.quotaExceeded){
+							storage_quota_exceeded();
+						}else{
+							// e.g. localStorage is disabled
+							// (or there's some other error?)
+						}
+					}
+				});
```

Comment:
There's no need to duplicate this code. You can extract it into a function instead.
Or actually, it's already a function below, as a convenient unit.
So just pull that out (preferably above both usages), and give it a name.
Something like `save_to_storage` or `save_image_to_storage`